### PR TITLE
Chore: Rewrite action-links to ts

### PR DESCRIPTION
**Pull request by KevLehman**

### CODE REVIEW

File: apps/meteor/app/action-links/server/lib/actionLinks.ts [hunk 1]
```diff
@@ -14,11 +15,11 @@ function getMessageById(messageId) {
 
 // Action Links namespace creation.
 export const actionLinks = {
-	actions: {},
-	register(name, funct) {
+	actions: {} as { [key: string]: Function },
```
Comment (sampaiodiego) [inline apps/meteor/app/action-links/server/lib/actionLinks.ts:18]: should we type this `Function` as well? not sure how much work that would add 😂 

I saw at least the first param is `IMessage`, but other idk

**Review by sampaiodiego — CHANGES_REQUESTED** [review]

File: apps/meteor/app/action-links/server/actionLinkHandler.ts [hunk 2]
```diff
@@ -11,7 +11,16 @@ Meteor.methods({
 
 		const message = actionLinks.getMessage(name, messageId);
 
-		const actionLink = message.actionLinks[name];
+		if (!message) {
+			throw new Meteor.Error('error-invalid-message', 'Invalid message', { method: 'actionLinkHandler' });
+		}
+
+		// NOTE: based on types (and how FE uses it) this should be the way of doing it
+		const actionLink = message.actionLinks && message.actionLinks.find((action) => action.method_id === name);
```
Comment (debdutdeb) [inline apps/meteor/app/action-links/server/actionLinkHandler.ts:19]:
```suggestion
		const actionLink = message.actionLinks?.find((action) => action.method_id === name);
```

File: apps/meteor/app/action-links/server/lib/actionLinks.ts [hunk 3]
```diff
@@ -1,8 +1,9 @@
+import { IMessage } from '@rocket.chat/core-typings';
 import { Meteor } from 'meteor/meteor';
 
 import { getMessageForUser } from '../../../../server/lib/messages/getMessageForUser';
 
-function getMessageById(messageId) {
+function getMessageById(messageId: string): IMessage | undefined {
```
Comment (debdutdeb) [inline apps/meteor/app/action-links/server/lib/actionLinks.ts:6]:
```suggestion
function getMessageById(messageId: IMessage['_id']): IMessage | undefined {
```

File: apps/meteor/app/action-links/server/lib/actionLinks.ts [hunk 4]
```diff
@@ -14,11 +15,11 @@ function getMessageById(messageId) {
...
 // Action Links namespace creation.
 export const actionLinks = {
-	actions: {},
-	register(name, funct) {
+	actions: {} as { [key: string]: Function },
+	register(name: string, funct: Function): void {
 		actionLinks.actions[name] = funct;
 	},
-	getMessage(name, messageId) {
+	getMessage(name: string, messageId: string): IMessage | undefined {
```
Comment (debdutdeb) [inline apps/meteor/app/action-links/server/lib/actionLinks.ts:22]:
```suggestion
	getMessage(name: string, messageId: IMessage['_id']): IMessage | undefined {
```

File: apps/meteor/app/action-links/server/lib/actionLinks.ts [hunk 5]
```diff
@@ -27,7 +28,7 @@ export const actionLinks = {
 			});
 		}
 
-		if (!message.actionLinks || !message.actionLinks[name]) {
+		if (!message.actionLinks || !message.actionLinks.some((action) => action.method_id === name)) {
```
Comment (debdutdeb) [inline apps/meteor/app/action-links/server/lib/actionLinks.ts:31]:
```suggestion
		if (!message.actionLinks?.some((action) => action.method_id === name)) {
```

**Review by sampaiodiego — APPROVED** [review]

File: apps/meteor/app/action-links/server/lib/actionLinks.ts (hunk 1 above)
Comment (KevLehman) [reply apps/meteor/app/action-links/server/lib/actionLinks.ts]: typed the func 👀 used the typings from the client file, but not really sure why `instance` is typed as undefined (and not sure what instance should be anyways)

**Pull request merged**